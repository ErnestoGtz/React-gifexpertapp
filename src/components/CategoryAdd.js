import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';


export const CategoryAdd = ({setCategories}) => {
    
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Submit hecho');
        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue,...categories]);
            setInputValue('');
        }
        
    }

    return (
        <Fragment>
            <form onSubmit={handleSubmit}>
                <input 
                    type="text"
                    value={inputValue}
                    onChange={ handleInputChange }
                />
            </form>
            
        </Fragment>
    )
}

CategoryAdd.propTypes = {
    setCategories:PropTypes.func.isRequired
}


