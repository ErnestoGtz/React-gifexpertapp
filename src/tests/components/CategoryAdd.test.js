import React from 'react';
import { shallow } from 'enzyme';
import { CategoryAdd } from '../../components/CategoryAdd';
import '@testing-library/jest-dom';

describe('Pruebas en el componente <CategoryAdd />', () => {
    //const setCategories = () => {};
    const setCategories = jest.fn();
    let wrapper = shallow(<CategoryAdd setCategories={setCategories} />);

    beforeEach(()=> {
        jest.clearAllMocks();
        wrapper = shallow(<CategoryAdd setCategories={setCategories} />);
    });

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change',{ target: {value:value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    });
    test('no debe de postear la informacion con submit ', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){} });
        expect(setCategories).not.toHaveBeenCalled();
    });
    test('debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'hola mundo';
        
        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){} });
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');
    });
    
    
});
