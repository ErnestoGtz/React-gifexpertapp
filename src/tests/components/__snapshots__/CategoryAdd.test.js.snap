// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CategoryAdd /> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
